.github/pull_request_template.md

```diff
@@ -0,0 +1,40 @@
+## Description
+<!-- Elaborate beyond the title of the PR as necessary to help the reviewers.-->
+
+## Related issues and pull requests
+<!-- List any issue that is fixed or related, such as documentation PR:
+
+* Fixes #issuenumber
+* Related documentation in #issuenumber
+-->
+
+## Documentation
+<!-- remove x to confirm that no documentation and/or no release notes entry is
+needed, or provide relevant information.-->
+
+- [x] User facing documentation needed
+- [x] Release notes entry needed
+
+## Release notes
+<!-- Add all needed release notes entries with section title, wording, and link
+  to related issue or pull request. Sections include General, Security,
+  JDBC driver, CLI, SPI, and connectors. If PR affects multiple sections,
+  especially also multiple connectors, list them all with the desired wording.
+  Use markdown syntax:
+
+## General
+
+* Improver performance of so and so. ({issue}`1234`)
+* Fix incorrect results .... type. ({issue}`5678`)
+
+## Hive connector
+
+* Add support for super cool feature. ({issue}`9123`)
+
+## Iceberg connector
+
+* Add support for super cool feature. ({issue}`91234`)
+
+Maintainers will comment for updates on wording during work on PR. Update in
+description as requested.
+-->
```

Comment:
This puts lot of burden on PR submitter to provide release notes entry. Maintainers can always edit the PR description anyway and add them there if we all are agreed that we'll collect release notes from the PR description rather than the "Release notes for XXX" issue we create today.

More useful content here would be to link to https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#commits-and-pull-requests maybe?

Reply:
> This puts lot of burden on PR submitter to provide release notes entry. Maintainers can always edit the PR description anyway and add them there if we all are agreed that we'll collect release notes from the PR description rather than the "Release notes for XXX" issue we create today.

i agree that this move of responsibility isn't what we want.

Reply:
I wasnt aware that maintainers can edit the description.. that make it easier. I will reword and add the link